### PR TITLE
Backport packing of palette indices from dav1d 1.3.0

### PR DESCRIPTION
**Pull request by fbossen**

### CODE REVIEW

File: src/decode.rs [hunk 1]
```diff
@@ -708,7 +708,8 @@ fn order_palette(
 unsafe fn read_pal_indices(
     ts: &mut Rav1dTileState,
     scratch_pal: &mut Rav1dTaskContext_scratch_pal,
-    pal_idx: &mut [u8],
+    pal_tmp: &mut [u8],
```
Comment (rinon) [inline src/decode.rs:711]: Would it make sense to pass a reference to `t.scratch` instead? @randomPoison has been working on this structure and might have an opinion.

File: src/decode.rs [hunk 2]
```diff
@@ -708,7 +708,8 @@ fn order_palette(
...
+    pal_idx: Option<&mut [u8]>, // if None, use pal_tmp instead of pad_idx
```
Comment (rinon) [inline src/decode.rs:712]: `pal_idx` instead of `pad_idx`?

File: src/decode.rs [hunk 3]
```diff
@@ -731,32 +732,52 @@ unsafe fn read_pal_indices(
...
         }
     }
-    if h4 < bh4 {
-        let y_start = h4 * 4;
-        let len = bw4 * 4;
-        let (src, dests) = pal_idx.split_at_mut(stride * y_start);
-        let src = &src[stride * (y_start - 1)..][..len];
-        for y in 0..(bh4 - h4) * 4 {
-            dests[y * stride..][..len].copy_from_slice(src);
+    if let Some(pal_idx) = pal_idx {
```
Comment (rinon) [inline src/decode.rs:754]: Could we just do the following and not duplicate the code?

```rust
let pal_idx = pal_idx.unwrap_or(pal_tmp);
...
```

File: src/decode.rs (hunk 2 above)
Comment (fbossen) [reply src/decode.rs]: Oops. Fixed it.

File: src/decode.rs (hunk 3 above)
Comment (fbossen) [reply src/decode.rs:754]: I had tried that, but ran into borrowing issues in `pal_idx[i] = pal_tmp[2 * i + 0] | (pal_tmp[2 * i + 1] << 4)` and ended up duplicating the code.

Comment (fbossen) [reply src/decode.rs:754]: Actually, the duplicated code will be put in a separate function (with asm versions) in a future PR, so this is a temporary issue

Comment (rinon) [reply src/decode.rs:746]: Sounds good, I see.

**Pull request merged**